phi/tools/discord_tools.py

```diff
@@ -0,0 +1,195 @@
+"""Discord integration tools."""
+
+import json
+from typing import Union
+from phi.tools import Toolkit
+from phi.utils.log import logger
+
+try:
+    import discord
+    from discord.ext import commands
+except ImportError:
+    raise ImportError("`discord` package not installed. Please install using `pip install discord.py`")
+
+MessageableChannel = Union[discord.TextChannel, discord.Thread, discord.DMChannel]
+
+
+class DiscordTools(Toolkit):
+    def __init__(
+        self,
+        bot_token: str,
+        enable_messaging: bool = True,
+        enable_history: bool = True,
+        enable_channel_management: bool = True,
+    ):
+        """Initialize Discord tools.
+
+        Args:
+            bot_token: Discord bot token for authentication
+            enable_messaging: Whether to enable message sending functionality
+            enable_history: Whether to enable message history retrieval
+            enable_channel_management: Whether to enable channel management
+        """
+        super().__init__(name="discord_tools")
+
+        self.bot_token = bot_token
```

Comment:
Can you also add get the bot_token from env. 
example - getenv("EXA_API_KEY")
Also add logger.error if bot token is not found.

Reply:
hey solved for this in the latest commit